feat: Add team member management components and actions

gentamura opened the pull request:
## Summary
This PR implements the team member management functionality, including invitation and listing features with proper role-based authorization.

## Related Issue
N/A

## Changes
- Add TeamMembersForm component for adding new team members
  - Email input with validation
  - Role selection (admin/member)
  - Form validation using valibot
  - Pro plan authorization check for invitation form
- Create server action for team member addition
  - Implement addTeamMember action with proper validation
  - Handle existing user verification
  - Implement team membership creation
  - Add admin role authorization checks
- Add type safety for TeamRole handling
  - Proper validation using valibot picklist
  - No type assertions for better type safety
- Refactor team fetching to use fetchCurrentTeam service
  - Handle subscription status for pro plan verification
  - Add proper type definitions for team data

## Testing
- Tested form validation for:
  - Email format validation
  - Role validation (admin/member)
  - Error message display
- Tested server-side validation and error handling:
  - User existence check
  - Duplicate team membership check
  - Role validation
  - Admin authorization checks
- Verified error messages are properly displayed to users
- Tested role management functionality
  - Role updates by admin users
  - Pro plan verification

## Other Information
- Uses Card component for consistent UI
- Implements proper error handling with user-friendly messages
- Invitation form only visible to Pro plan teams

gentamura commented:
@satococoa review, please :pray:

satococoa reviewed (changes requested):
Thank you for your implementation!!

Now, Please could you add these two functionalities?

- Only Pro team can view invitation form.
- Only Admin user can invite(manage) members.

gentamura commented:
@satococoa 

Implemented missing features:
> * Only Pro team can view invitation form.
> * Only Admin user can invite(manage) members.

Also refactored to use fetchCurrentTeam service method instead of getTeam method.

Please review again :pray:

satococoa commented on app/(main)/settings/team/team-members.tsx:
```diff
@@ -0,0 +1,15 @@
+
+	return (
+		<Card title="Team members">
+			{isProPlan(team) && teamRole === "admin" && <TeamMembersForm />}
```
👍

satococoa commented on app/(main)/settings/team/actions.ts:
```diff
@@ -44,3 +73,135 @@ export async function updateTeamName(teamDbId: number, formData: FormData) {
+			throw new Error("Invalid role");
+		}
+
+		// 1. Get current user's team
```
Can you use `fetchCurrentTeam()` here?

gentamura replied:
7cb2e5a8d10b fixed🧑‍🔧

satococoa commented on app/(main)/settings/team/actions.ts:
```diff
@@ -44,3 +73,135 @@ export async function updateTeamName(teamDbId: number, formData: FormData) {
+		}
+
+		const currentTeam = team[0]; // TODO: This will need to be adjusted after implementing team switching
+
```
Since Server actions must be treated as Public endpoint, please validate these, here:

- Current Team is a Pro plan (including internal) team.
- Current User is an Admin role user.

ref
https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#security

gentamura replied:
56434bca8a6b fixed ✍️

satococoa reviewed (commented):
Thank you for the fixes and refactoring.
I really appreciate that.

I've confirmed that validations have been added to the UI properly .
However, I think we should also implement these validations on the server side.

satococoa commented on app/(main)/settings/team/actions.ts:
```diff
@@ -1,10 +1,39 @@
+	return role === "admin" || role === "member";
+}
+
+export async function getTeamName() {
+	const user = await getUser();
+
+	// TODO: In the future, this query will be changed to retrieve from the selected team ID
+	const _teams = await db
+		.select({ dbId: teams.dbId, name: teams.name })
+		.from(teams)
+		.innerJoin(teamMemberships, eq(teams.dbId, teamMemberships.teamDbId))
+		.innerJoin(
+			supabaseUserMappings,
+			eq(teamMemberships.userDbId, supabaseUserMappings.userDbId),
+		)
+		.where(eq(supabaseUserMappings.supabaseUserId, user.id));
+
+	return _teams[0].name;
+}
+
```
This can be removed.

```suggestion
```

I've removed this on my pull request.
https://github.com/giselles-ai/giselle/pull/163/files#diff-fd9634083caa6efe56527cb4a5c08420cc90650a15640a08396d234d38c29eebL14

gentamura replied:
[9cf6781](https://github.com/giselles-ai/giselle/pull/135/commits/9cf6781aaf8f150462c3b2aade9dc695614db330) 
removed ✂️

satococoa commented on app/(main)/settings/team/actions.ts:
```diff
@@ -44,3 +73,131 @@ export async function updateTeamName(teamDbId: number, formData: FormData) {
+		// 1. Get current team
+		const currentTeam = await fetchCurrentTeam();
+		if (!isProPlan(currentTeam)) {
+			throw new Error("Only pro plan teams can add members");
```
👍

satococoa commented on app/(main)/settings/team/actions.ts:
```diff
@@ -44,3 +73,131 @@ export async function updateTeamName(teamDbId: number, formData: FormData) {
+			!currentUserRoleResult.success ||
+			currentUserRoleResult.data !== "admin"
+		) {
+			throw new Error("Only admin users can add team members");
```
👍

satococoa reviewed (approved):
👍 Thank you!

gentamura commented:
@satococoa 
In the meantime, we have also adjusted the following commitments :pray:
f031aebd8251

satococoa commented:
@gentamura 

> In the meantime, we have also adjusted the following commitments 🙏

Oops, I got ahead of myself. Sorry about that!

gentamura commented:
@shige @toyamarinyon Please review :pray:

shige reviewed (approved):
LGTM! 🔥

gentamura commented:
Thanks for reviewers 🚀
